packages/interfaces/src/message.ts

```diff
@@ -1,5 +1,10 @@
+import { ShardInfo } from "./enr.js";
 import type { PubsubTopic } from "./misc.js";
 
+export interface SingleTopicShardInfo extends Omit<ShardInfo, "indexList"> {
```

Comment:
```suggestion
export interface SingleTopicShardInfo extends Omit<ShardInfo, "indexList"> {
```

Topic and Shard are redundant. It's a single shard info, which is also a single topic info.

The redundancy of information makes the interface name more complicated for no added value.

Also, you break the interface with `ShardInfo` here as you removed the index list.

Considering you only bring one property (`cluster`) over from `ShardInfo`, I am not sure to understand the point of extending from `ShardInfo`. Just create a new type.

Re-using the type is only useful if you want a function that accept `ShardInfo` to also accept `SingleTopicShardInfo`. But because of the `Omit`, you are not getting that.

If that's what you want, then try to override the `indexList` with a [tuple type](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types) to specify that it's an array with only one `number`.

If you don't need interface compatibility between `ShardInfo` and `SingleTopicShardInfo`, then don't make the latter extend the former, just redefine a new interface.

Reply:
IMO `Omit` makes sense when you want common API to be used. e.g.

```
interface A {
foo: number
bar: string
}

interface B extends <A, "foo"> {}

function func (alpha: Partial<A>) {
...
}

// this works
func(B)
```